src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -409,84 +408,73 @@ VideoDecoder::VideoStreamOptions::VideoStreamOptions(
   }
 }
 
-void VideoDecoder::addVideoStream(
+void VideoDecoder::addStream(
     int streamIndex,
-    const VideoStreamOptions& videoStreamOptions) {
+    AVMediaType mediaType,
+    const torch::Device& device) {
   TORCH_CHECK(
       activeStreamIndex_ == NO_ACTIVE_STREAM,
       "Can only add one single stream.");
+  TORCH_CHECK(
+      mediaType == AVMEDIA_TYPE_VIDEO || mediaType == AVMEDIA_TYPE_AUDIO,
+      "Can only add video or audio streams.");
   TORCH_CHECK(formatContext_.get() != nullptr);
 
   AVCodecOnlyUseForCallingAVFindBestStream avCodec = nullptr;
 
   activeStreamIndex_ = av_find_best_stream(
-      formatContext_.get(), AVMEDIA_TYPE_VIDEO, streamIndex, -1, &avCodec, 0);
+      formatContext_.get(), mediaType, streamIndex, -1, &avCodec, 0);
+
   if (activeStreamIndex_ < 0) {
-    throw std::invalid_argument("No valid stream found in input file.");
+    throw std::invalid_argument(
+        "No valid stream found in input file. Is " +
+        std::to_string(streamIndex) + " of the desired media type?");
```

Comment:
Minor, but since we're changing this, we could make it a `TORCH_CHECK`. We have a mix of `TORCH_CHECK` and throwing exceptions; I think we should make them all `TORCH_CHECK()`s. It's not a big enough deal to do everywhere, but might as well as we change code. Heads-up that it might require changing the C++ test case; it might expect `std::invalid_argument`.

Reply:
Agreed, let me open an issue for this, this would be a good first contribution